src/rotest/api/resource_control/release_resources.py

```diff
@@ -0,0 +1,111 @@
+# pylint: disable=unused-argument, no-self-use
+import httplib
+
+from django.db import transaction
+from django.core.exceptions import ObjectDoesNotExist
+from swaggapi.api.builder.server.response import Response
+from swaggapi.api.builder.server.request import DjangoRequestView
+
+from rotest.management import ResourceData
+from rotest.management.common.utils import get_username
+from rotest.common.django_utils.common import get_sub_model
+from rotest.api.common.models import ReleaseResourcesParamsModel
+from rotest.api.common.responses import (BadRequestResponseModel,
+                                         SuccessResponse)
+from rotest.management.common.errors import (ResourceAlreadyAvailableError,
+                                             ResourceDoesNotExistError,
+                                             ResourcePermissionError,
+                                             ResourceReleaseError,
+                                             ServerError)
+
+
+class ReleaseResources(DjangoRequestView):
+    """Release the given resources one by one.
+
+    For complex resource, marks also its sub-resources as free.
+
+    Raises:
+        ResourceReleaseError: if resource is a complex resource and fails.
+        ResourcePermissionError: if resource is locked by other user.
+        ResourceAlreadyAvailableError: if resource was already available.
+    """
+    URI = "resources/release_resources"
+    DEFAULT_MODEL = ReleaseResourcesParamsModel
+    DEFAULT_RESPONSES = {
+        httplib.NO_CONTENT: SuccessResponse,
+        httplib.BAD_REQUEST: BadRequestResponseModel
+    }
+    TAGS = {
+        "post": ["Resources"]
+    }
+
+    def _release_resource(self, resource, user_name):
+        """Mark the resource as free.
+
+        For complex resource, marks also its sub-resources as free.
+
+        Args:
+            resource (ResourceData): resource to release.
+            user_name (str): name of the releasing user.
+
+        Raises:
+            ResourceReleaseError: if resource is a complex resource and fails.
+            ResourcePermissionError: if resource is locked by other user.
+            ResourceAlreadyAvailableError: if resource was already available.
+        """
+        errors = {}
+
+        for sub_resource in resource.get_sub_resources():
+            try:
+                self._release_resource(sub_resource, user_name)
+
+            except ServerError as ex:
+                errors[sub_resource.name] = (ex.ERROR_CODE, str(ex))
```

Comment:
I think you're not including all of the traceback
Make sure

Reply:
i am not sure what u meant?
the exception except statement?
or the error caught?
both of them are a copy paste  from the original code..
i am not sure what to do then.